uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/AzureSampleApp.java

```diff
@@ -128,13 +143,16 @@ public void signOutFromSingleAccountFragment() {
      */
     public void confirmSignedIn(@NonNull final String username) {
         Logger.i(TAG, "Confirming account with supplied username is signed in..");
+
+        final UiObject signedInUser = UiAutomatorUtils.obtainUiObjectWithResourceId("com.azuresamples.msalandroidapp:id/current_user");
+
+        // Sometimes need some time for upn to appear
```

Comment:
10 secs seem high, does it happen in some specific scenarios?

Reply:
No, it's basically to give azure sample app time to show upn as it's starting up. We can go about this a different way. Instead of finding the ui object and checking it's text, which leaves us open to checking it too quick before it loads. We can instead just search for the username text directly (or "none" if no account is logged in), and this will allow us to pass the test as soon as the username becomes apparent. I will do some testing for this.

Reply:
Implemented these changes.